### ISSUE TITLE
[Question] How leaf output determined for LambdaMART?

### ISSUE THREAD

**Issue by dszhengyu** [microsoft/LightGBM#2283]
Hi, 

I'm trying to train a learning to rank model with GBDT and the customized loss function(by assigned the fobj), so I dive deep into the code base of LightGBM to get more clear how things work for my case. 

I figured out almost everything except how the leaf value is calculated. 
Per my understanding, the leaf output should be computed as (-gradients/hessians) * learning_rate, but I could no found the code snippets like that.

Is my understanding right? Could you help my locate the code? 

Thx
-Yu

**Comment by guolinke** [comment]
https://github.com/microsoft/LightGBM/blob/ee28ea36a456a064bc1d6ea900a0880f9a3c711f/src/treelearner/feature_histogram.hpp#L451-L458

**Comment by dszhengyu** [comment]
Thx a lot!
